src/DataStructures/DataBox/Prefixes.hpp

```diff
@@ -43,7 +37,7 @@ struct dt : db::PrefixTag, db::SimpleTag {
 /// \snippet Test_DataBoxPrefixes.cpp analytic_name
 template <typename Tag>
 struct Analytic : db::PrefixTag, db::SimpleTag {
-  using type = db::const_item_type<Tag>;
+  using type = typename Tag::type;
```

Comment:
@kidder What's the reasoning for eliminating `db::const_item_type`? You don't write that in the description of the PR. Is it related to compile time?

`db::const_item_type` currently handles the return type for compute tags. Using `Tag::type` removes that logic. For example, `Tags::Analytic<SomeComputeTag>` won't work anymore, which of course is an edge case, but I'm not clear on what we gain by removing this logic.

Reply:
Compute tags are all switched over to mutating compute tags. They currently use `return_type` to specify their type, but this is being eliminated. Since compute tags in general should inherit off of a simple tag, they inherit a `::type`, which better be the type they are computing.

The only edge case is, as you point out, is that pointers are derefenced. I view this as another reason to avoid use `::type` or any of the `item_type` aliases. Not only do they make the code more difficult to read, they also make the code more difficult to maintain. Having said that, @kidder has already done a lot of heroic grunt work on this front and so I don't think it's fair to ask him to do more. Everyone should be updating these types of things as they look at files in order to spread the workload out.

Reply:
Oh, `db::get` still does the dereferencing, so if we replace `Tag::type::element_type` by a generic template parameter then the function doesn't depend anymore on whether or not the tag stores a pointer, correct? That seems fine then!

Going forward, should we establish the guideline that functions should take explicit types where possible, and else use template parameters (even where the tags are available)? With C++20 we can even [switch those to `auto`](https://en.cppreference.com/w/cpp/language/function_template#Abbreviated_function_template). That means we should gradually (i.e. spreading out the workload as you said) switch most/all of the `Tag::type`s that Larry adds in this PR to template parameters? I wonder if it would be easier to do that in this PR so people don't start copying code that we have to ask them to change in code review, although I don't want to ask Larry to do this work alone of course..